torch_ort_inference/torch_ort_inference/ortinferencemodule/ortinferencemodule.py

```diff
@@ -0,0 +1,255 @@
+            debug_options = DebugOptions()
+
+        if not provider_options:
+            provider_options = ProviderOptions()
+
+        try:
+            super(ORTInferenceModule, self).__init__()
+            self._original_module = module
+            utils.patch_ortinferencemodule_forward_method(self)
+            self._flattened_module = _io._FlattenedModule(module)
+            self._debug_options = debug_options
+            # onnx models
+            self._onnx_models = _onnx_models.ONNXModels()
+            self._module_parameters = list(inspect.signature(self._original_module.forward).parameters.values())
+            self._device = utils.get_device_from_module(module)
+            self._export_mode = torch.onnx.TrainingMode.EVAL
+            self._export_extra_kwargs = {}
+            self._provider_options = provider_options
+            self._graph_initializers = None
+            self._graph_info = None
+
+        except Exception as e:
+            raise e
+
+        # Finally, ORTInferenceModule initialization is complete.
+        self._is_initialized = True
+
+    def forward_call(self, *inputs, **kwargs):
+        """Delegate the :meth:`~torch.nn.Module.forward` to
+        ONNX Runtime.
+
+        The first call to forward performs setup and checking steps. During this call,
+        ORTInferenceModule determines whether the module can be trained with ONNX Runtime. For
+        this reason, the first forward call execution takes longer than subsequent calls.
+        Execution is interrupted if ONNX Runtime cannot process the model for inferencing.
+
+        Args:
+            *inputs and **kwargs represent the positional, variable positional, keyword
+            and variable keyword arguments defined in the user's PyTorch module's forward
+            method. Values can be torch tensors and primitive types.
+
+        Returns:
+            The output as expected from the forward method defined by the user's
+            PyTorch module. Output values supported include tensors, nested sequences
+            of tensors and nested dictionaries of tensor values.
+        """
+        schema = _io._extract_schema({"args": copy.copy(inputs), "kwargs": copy.copy(kwargs)})
+        if (
+            self._onnx_models.exported_model
+            and schema == self._input_info.schema
+        ):
+            # All required models have already been exported previously
+            pass
+        else:
+            # Exporting module to ONNX for the first time
+            self._set_device_from_module(inputs, kwargs)
+            self._onnx_models.exported_model = self._export_model(schema,*inputs, **kwargs)
+            if self._debug_options.save_onnx_models.save:
+                self._onnx_models.save_exported_model(
+                    self._debug_options.save_onnx_models.path,
+                    self._debug_options.save_onnx_models.name_prefix,
+                    self._export_mode,
+                )
+
+        try:
```

Comment:
Again, this try block is not needed here since you're catching everything anyway and raising the exception. Please remove this.

Reply:
Noted. PR updated.